compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java

```diff
@@ -164,6 +171,58 @@ public static boolean isValidPng(Path filePath) throws IOException {
         return isMatchingImageFormat(filePath, PNG_HEX_HEADER, 8);
     }
 
+    /**
+     * Get last modified timestamp of a ballerina project.
+     *
+     * @param projectRoot project root path
+     * @return last modified time of the ballerina project
+     */
+    public static long lastModifiedTimeOfBalProject(Path projectRoot) {
+        File[] files = projectRoot.toFile().listFiles();
+        long latestDate = 0;
+        if (files != null) {
+            for (File file : files) {
+                long fileModifiedDate = latestDate;
+                String filename = file.toPath().getFileName().toString();
+                if (file.isDirectory()) {
+                    if (isChild(projectRoot.resolve(MODULES_ROOT), file.toPath())
+                            && isChild(projectRoot.resolve(TEST_DIR_NAME), file.toPath())
+                            && isChild(projectRoot.resolve(RESOURCE_DIR_NAME), file.toPath())) {
+                        fileModifiedDate = lastModifiedTimeOfBalProject(file.toPath());
+                    }
+                } else {
+                    if (file.toPath().equals(projectRoot.resolve(BALLERINA_TOML))) {
+                        // Ballerina.toml
+                        fileModifiedDate = file.lastModified();
+                    } else if (filename.endsWith(BLANG_SOURCE_EXT)
+                            && file.toPath().equals(projectRoot.resolve(filename))) {
+                        // default module ballerina source files
+                        fileModifiedDate = file.lastModified();
+                    } else if (isChild(projectRoot.resolve(MODULES_ROOT), file.toPath())
+                            && filename.endsWith(BLANG_SOURCE_EXT)) {
```

Comment:
We don't need to check each `.bal` file inside modules. This is the list of files/directories we should check:
1. Following locations in the package root directory
    i.  All `.bal` files
    ii. `resources/`
    iii. `tests/`
    iv. `Ballerina.toml`
2. `<packageRoot>/modules` directory

Reply:
Pls check the revised logic